velox/common/memory/tests/MockSharedArbitratorTest.cpp

```diff
@@ -1802,37 +1811,74 @@ DEBUG_ONLY_TEST_F(
 }
 
 TEST_F(MockSharedArbitrationTest, singlePoolGrowWithoutArbitration) {
-  const int64_t memoryCapacity = 128 << 20;
-  const uint64_t memoryPoolInitCapacity = 32 << 20;
-  const uint64_t memoryPoolTransferCapacity = 8 << 20;
-  setupMemory(
-      memoryCapacity, 0, memoryPoolInitCapacity, 0, memoryPoolTransferCapacity);
+  int64_t memoryCapacity = 512 << 20;
+  uint64_t memoryPoolInitCapacity = 32 << 20;
+  struct TestParam {
+    uint64_t memoryPoolTransferCapacity;
+    uint64_t fastExponentialGrowthCapacityLimit;
+    double slowCapacityGrowPct;
+  };
 
-  auto* memOp = addMemoryOp();
-  const int allocateSize = 1 * MB;
-  while (memOp->capacity() < memoryCapacity) {
-    memOp->allocate(allocateSize);
-  }
+  // Try to make each test allocation larger than the largest memory pool
+  // quantization(8MB) to not have noise.
+  std::vector<TestParam> testParams{
+      {8 << 20, 0, 0},
+      {0, 128 << 20, 0.1},
+      {32 << 20, 128 << 20, 0.1},
+      {16 << 20, 128 << 20, 0.5},
+  };
 
-  verifyArbitratorStats(
-      arbitrator_->stats(),
-      memoryCapacity,
-      0,
-      0,
-      (memoryCapacity - memoryPoolInitCapacity) / memoryPoolTransferCapacity);
+  for (const auto& testParam : testParams) {
+    setupMemory(
+        memoryCapacity,
+        0,
+        memoryPoolInitCapacity,
+        0,
+        testParam.memoryPoolTransferCapacity,
+        testParam.fastExponentialGrowthCapacityLimit,
+        testParam.slowCapacityGrowPct);
 
-  verifyReclaimerStats(
-      memOp->reclaimer()->stats(),
-      0,
-      (memoryCapacity - memoryPoolInitCapacity) / memoryPoolTransferCapacity);
+    auto* memOp = addMemoryOp();
+    const int allocateSize = 1 * MB;
+    while (memOp->capacity() < memoryCapacity) {
+      memOp->allocate(allocateSize);
+    }
 
-  clearTasks();
-  verifyArbitratorStats(
-      arbitrator_->stats(),
-      memoryCapacity,
-      memoryCapacity,
-      0,
-      (memoryCapacity - memoryPoolInitCapacity) / memoryPoolTransferCapacity);
+    uint64_t expectedNumRequests{0};
+    if (testParam.fastExponentialGrowthCapacityLimit == 0) {
```

Comment:
Can we make expectedNumRequests as a test parameter? We can comment about how we get expectedNumRequests? Thanks!

Reply:
My thoughts were it would be easier to present the calculation of expected number of requests for later debugging. I can put additional comments above this section.